### PR TITLE
docs: create 'concepts' page on scopes

### PR DESCRIPTION
**Pull request by ahal**
[Rendered](https://taskcluster-taskgraph--358.org.readthedocs.build/en/358/concepts/scopes.html)

### CODE REVIEW

**Comment by ahal** [comment]
@matt-boris @petemoore @lotas hey TC team, I wrote up this doc because I realized that even I was feeling fuzzy on how scopes actually work under the hood. Could one of you do a quick proof read of the `Queue and Auth Services` section just to make sure I have it more or less right?

File: docs/concepts/scopes.rst [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+Given the requested scopes in a task definition and the access control list in
+the deployment configuration, Taskcluster can now make a decision about whether
+a task is authorized to use the scopes it requests or not. This is where the `queue`_
+and `auth`_ services come in.
+
+The queue service:
+
+1. Inspects the requested scopes in the task's definition.
+2. `Queries the auth service`_ to get the current set of expanded scopes.
```
[the remark this answers is not in the document]
Comment (ahal) [reply docs/concepts/scopes.rst:92]: Maybe https://docs.taskcluster.net/docs/reference/platform/auth/roles#roles-and-role-expansion ?

[the remark this answers is not in the document]
Comment (bhearsum) [reply docs/concepts/scopes.rst:92]: Ah, yes! (No wonder it didn't come up when I searched for `expand`...)

**Comment by ahal** [comment]
Thanks for the suggestions, I incorporated them in!

**Pull request merged**